### PR TITLE
Defect/de3716 show all leaders

### PR DESCRIPTION
**Pull request by Doug-Shannon**
This is a small refactor to the participant service and gathering component in an effort to clean up the code a bit.  Adds a lot of functionality to the participant service for checking a users role, or getting a group of users with a certain role.

### CODE REVIEW

File: src/app/components/pin-details/gathering/gathering.component.spec.ts [hunk 1]
```diff
@@ -204,47 +207,49 @@ describe('GatheringComponent', () => {
...
     });
 
     it('should init and fail to get full address then toast', () => {
         let pin = MockTestData.getAPin(1);
         let participants = MockTestData.getAParticipantsArray(3);
         let expectedText = '<p>Looks like there was an error. Please fix and try again</p>';
         mockContentService.getContent.and.returnValue(expectedText);
-        mockSessionService.getContactId.and.returnValue(participants[2].contactId);
         mockParticipantService.getParticipants.and.returnValue(Observable.of(participants));
+        (<jasmine.Spy>mockParticipantService.getCurrentUserGroupRole).and.returnValue(Observable.of(GroupRole.LEADER));
```
Comment (JoeKer1) [inline src/app/components/pin-details/gathering/gathering.component.spec.ts]: Still don't need <jasmine.Spy> here :)

File: src/app/components/pin-details/gathering/gathering.component.ts [hunk 2]
```diff
@@ -71,56 +74,55 @@ export class GatheringComponent implements OnInit {
...
-  private loggedInUserIsInGathering(contactId: number) {
-    return this.pin.gathering.Participants.find((participant) => {
-      return (participant.contactId === contactId);
-    });
-  }
-
   public requestToJoin() {
     let successBodyContentBlock: string = this.app.isConnectApp() ? 'finderGatheringJoinRequestSent' : 'finderGroupJoinRequestSent';
-    this.angulartics2.eventTrack.next({ action: 'Join Gathering Button Click', properties: { category: 'Connect' }});
+    this.angulartics2.eventTrack.next({ action: 'Join Gathering Button Click', properties: { category: 'Connect' } });
```
Comment (JoeKer1) [inline src/app/components/pin-details/gathering/gathering.component.ts]: Eventually, we need to put the category based on the app mode probably?

File: src/app/components/pin-details/gathering/gathering.component.ts [hunk 3]
```diff
@@ -56,8 +58,9 @@ export class GatheringComponent implements OnInit {
     private listHelperService: ListHelperService,
     private content: ContentService,
     private angulartics2: Angulartics2,
-    public appSettingsService: AppSettingsService) {}
+    public appSettingsService: AppSettingsService) { }
```
Comment (vchtelmakh) [inline src/app/components/pin-details/gathering/gathering.component.ts]: Why the space?

**Comment by JoeKer1** [comment]
is getIsCurrentUserALeader in participant service tested?

**Pull request merged**